stacker/actions/base.py

```diff
@@ -53,7 +53,7 @@ def build_walker(concurrency):
     """
     if concurrency == 1:
         return walk
-    return ThreadedWalker(concurrency).walk
+    return ThreadedWalker(threading.Semaphore(concurrency)).walk
```

Comment:
I think this change would break the default behavior, for when `concurrency == 0`. When it's 0, `ThreadedWalker` would default to an `UnlimitedSemaphore`: https://github.com/cloudtools/stacker/blob/cfe719e1434db36db7322d9927a8cc85dda8a24b/stacker/dag/__init__.py#L424

But with this change as is, it'd use a `threading.Semaphore(0)`.

I think it'd be better to just bubble that logic up into `build_walker` so it's all together:

```python
if concurrency == 1:
    return walk

semaphore = UnlimitedSemaphore()
if concurrency > 1:
    semaphore = threading.Semaphore(concurrency)

return ThreadedWalker(semaphore).walk
```